[7.9.x] Fix expanded ids for GScan

kinow opened the pull request:
These changes close #3281

I am looking at the old GScan to understand how it creates the states + states totals (related to https://github.com/cylc/cylc-ui/issues/401), and also want to understand how Cylc 8's `cylc monitor` is doing that.

While doing it I remembered there was a bug in GScan regarding expanded rows. And since I had some experience with Cylc UI's tree and expanded/collapsed states decided to check if there was a simple fix. Took 20 mins after attaching a debugger to find one possible solution.

Not sure if the best one as I don't have a lot of experience with the old GUI, but just in case that's helpful.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (why? manual testing only should do... we don't have many tests for the old GUI).
- [x] Appropriate change log entry included.
- [x] No documentation update required.

kinow commented:
The ID's used at the moment in 7.8.x:

![image](https://user-images.githubusercontent.com/304786/76175792-1128ab00-6213-11ea-8230-a8fe4b36d9b5.png)

kinow commented:
Testing:

1. When you expand one item out of the two, if you click on the "Refresh" button the issue happens once it's refreshed.
2. I tested with two suites, with all possible states I could think. Should work with remote suites too as the ID would have a different host, so hopefully no regressions :+1: 

![2020-03-09 14-39-08 flv](https://user-images.githubusercontent.com/304786/76176002-f0ad2080-6213-11ea-9500-0b894f95da20.gif)

hjoliver commented:
@kinow - this seems to work, except in one case: if I add the "groups" column (by the View menu) that adds another tree level at the top, and all suites underneath get auto-expanded again on update or refresh.

kinow commented:
Thanks for testing it @hjoliver . Had never seen that group column, but debugging it, I think the row ID should be composed of (group, host, user, suite).

Otherwise, if we use only (host, user, suite), we will get rows with (None, None, None), for example the row that contains only the group name, and no host/user/suite information. It looks like when we tell GTK that the row (None, None, None) is expanded, it expands all rows (?)

Updated the PR with a function with the code to fetch row ID's, to avoid code duplication, and added a couple lines of comments. Tested and appears to be working now. Could you try it again, please @hjoliver ?

If you confirm it worked, I will cherry-pick the commits to the 7.8.x branch/PR.

hjoliver commented:
Confirmed as working. Thanks @kinow :+1:

kinow commented:
Thanks @hjoliver ! Cherry-picked to the other branch too.
